### PR TITLE
add translated expressions help for map_report

### PR DESCRIPTION
**Pull request by Gustry**
### What does it fix?
* Funded by: DFAT
* Description: We can now add some translated help text and also some HTML.

@timlinux Can you test on your compiled QGIS 2.18? It has to be QGIS >= 2.18.13
I'm doing this PR on blind mode, I can't really preview the result in QGIS.

### Checklist:
- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Add to the changelog in metadata.txt if it's a new feature
- [ ] Unit test for new code added
- [x] Request someone to review or test your PR

### CODE REVIEW

File: safe/report/expressions/map_report.py [hunk 1]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=legend_title_header_element_help,
+    helpText=legend_title_header_element_help)
 def legend_title_header_element(feature, parent):
     """Retrieve legend title header string from definitions."""
     _ = feature, parent  # NOQA
     header = legend_title_header['string_format']
     return header.capitalize()
 
+disclaimer_title_header_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 2]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=disclaimer_title_header_element_help,
+    helpText=disclaimer_title_header_element_help)
 def disclaimer_title_header_element(feature, parent):
     """Retrieve disclaimer title header string from definitions."""
     _ = feature, parent  # NOQA
     header = disclaimer_title_header['string_format']
     return header.capitalize()
 
+disclaimer_text_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 3]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=disclaimer_text_element_help,
+    helpText=disclaimer_text_element_help)
 def disclaimer_text_element(feature, parent):
     """Retrieve disclaimer text string from definitions."""
     _ = feature, parent  # NOQA
     text = disclaimer_text['string_format']
     return text
 
+information_title_header_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 4]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=information_title_header_element_help,
+    helpText=information_title_header_element_help)
 def information_title_header_element(feature, parent):
     """Retrieve information title header string from definitions."""
     _ = feature, parent  # NOQA
     header = information_title_header['string_format']
     return header.capitalize()
 
+time_title_header_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 5]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=time_title_header_element_help,
+    helpText=time_title_header_element_help)
 def time_title_header_element(feature, parent):
     """Retrieve time title header string from definitions."""
     _ = feature, parent  # NOQA
     header = time_title_header['string_format']
     return header.capitalize()
 
+caution_title_header_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 6]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=caution_title_header_element_help,
+    helpText=caution_title_header_element_help)
 def caution_title_header_element(feature, parent):
     """Retrieve caution title header string from definitions."""
     _ = feature, parent  # NOQA
     header = caution_title_header['string_format']
     return header.capitalize()
 
+caution_text_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 7]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=caution_text_element_help,
+    helpText=caution_text_element_help)
 def caution_text_element(feature, parent):
     """Retrieve caution text string from definitions."""
     _ = feature, parent  # NOQA
     text = caution_text['string_format']
     return text
 
+source_title_header_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 8]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=source_title_header_element_help,
+    helpText=source_title_header_element_help)
 def source_title_header_element(feature, parent):
     """Retrieve source title header string from definitions."""
     _ = feature, parent  # NOQA
     header = source_title_header['string_format']
     return header.capitalize()
 
+analysis_title_header_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 9]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=analysis_title_header_element_help,
+    helpText=analysis_title_header_element_help)
 def analysis_title_header_element(feature, parent):
     """Retrieve analysis title header string from definitions."""
     _ = feature, parent  # NOQA
     header = analysis_title_header['string_format']
     return header.capitalize()
 
+version_title_header_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 10]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=version_title_header_element_help,
+    helpText=version_title_header_element_help)
 def version_title_header_element(feature, parent):
     """Retrieve version title header string from definitions."""
     _ = feature, parent  # NOQA
     header = version_title_header['string_format']
     return header.capitalize()
 
+reference_title_header_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 11]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=reference_title_header_element_help,
+    helpText=reference_title_header_element_help)
 def reference_title_header_element(feature, parent):
     """Retrieve reference title header string from definitions."""
     _ = feature, parent  # NOQA
     header = reference_title_header['string_format']
     return header.capitalize()
 
+crs_text_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 12]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=unknown_source_text_element_help,
+    helpText=unknown_source_text_element_help)
 def unknown_source_text_element(feature, parent):
     """Retrieve reference title header string from definitions."""
     _ = feature, parent  # NOQA
     header = unknown_source_text['string_format']
     return header.capitalize()
 
+aggregation_not_used_text_element_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 13]
```diff
@@ -31,164 +31,267 @@
...
+    args='auto', group=label_group, usesGeometry=False, referencedColumns=[],
+    help_text=aggregation_not_used_text_element_help,
+    helpText=aggregation_not_used_text_element_help)
 def aggregation_not_used_text_element(feature, parent):
     """Retrieve reference title header string from definitions."""
     _ = feature, parent  # NOQA
     header = aggregation_not_used_text['string_format']
     return header.capitalize()
 
+inasafe_logo_black_path_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 14]
```diff
@@ -31,164 +31,267 @@
...
-    args='auto', group=image_group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=image_group, usesGeometry=False, referencedColumns=[],
+    help_text=inasafe_logo_black_path_help,
+    helpText=inasafe_logo_black_path_help)
 def inasafe_logo_black_path(feature, parent):
     """Retrieve the full path of inasafe-logo-black.svg."""
     _ = feature, parent  # NOQA
     return black_inasafe_logo_path['path']
 
+inasafe_logo_white_path_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 15]
```diff
@@ -31,164 +31,267 @@
...
-    args='auto', group=image_group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=image_group, usesGeometry=False, referencedColumns=[],
+    help_text=inasafe_logo_white_path_help,
+    helpText=inasafe_logo_white_path_help)
 def inasafe_logo_white_path(feature, parent):
     """Retrieve the full path of inasafe-logo-white.svg."""
     _ = feature, parent  # NOQA
     return white_inasafe_logo_path['path']
 
+north_arrow_path_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/map_report.py [hunk 16]
```diff
@@ -31,164 +31,267 @@
...
-    args='auto', group=image_group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=image_group, usesGeometry=False, referencedColumns=[],
+    help_text=north_arrow_path_help,
+    helpText=north_arrow_path_help)
 def north_arrow_path(feature, parent):
     """Retrieve the full path of default north arrow logo."""
     _ = feature, parent  # NOQA
     return inasafe_north_arrow_path['path']
 
+organisation_logo_path_help = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/map_report.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/gis/generic_expressions.py [hunk 17]
```diff
@@ -76,9 +100,19 @@ def inasafe_place_value_name(number, feature, parent):
     else:
         return unit['name']
 
+description = tr(
```
Comment (ismailsunni) [inline safe/gis/generic_expressions.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/gis/generic_expressions.py [hunk 18]
```diff
@@ -101,9 +135,23 @@ def inasafe_place_value_coefficient(number, feature, parent):
     else:
         return None
 
+description = tr(
```
Comment (ismailsunni) [inline safe/gis/generic_expressions.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/gis/generic_expressions.py [hunk 19]
```diff
@@ -123,9 +171,19 @@ def inasafe_place_value_percentage(number, total, feature, parent):
     percentage = round((float(number) / float(total)) * 100, 1)
     return percentage_format.format(percentage=percentage)
 
+description = tr(
```
Comment (ismailsunni) [inline safe/gis/generic_expressions.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/gis/generic_expressions.py [hunk 20]
```diff
@@ -140,9 +198,19 @@ def beautify_date(inasafe_time, feature, parent):
     date = datetime_object.strftime('%Y-%m-%d')
     return date
 
+description = tr(
```
Comment (ismailsunni) [inline safe/gis/generic_expressions.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 21]
```diff
@@ -31,23 +32,45 @@
...
 @qgsfunction(
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def inasafe_logo_white_path(feature, parent):
     """Retrieve the full path of inasafe-logo-white.svg."""
     _ = feature, parent  # NOQA
     return inasafe_logo_white['path']
 
+description = tr('Retrieve a header name of the field name from definitions.')
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 22]
```diff
@@ -77,11 +100,22 @@ def inasafe_field_header(field, feature, parent):
             return name.capitalize()
     return None
 
+description = tr('Retrieve units of the given minimum needs field name.')
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 23]
```diff
@@ -115,11 +149,22 @@ def minimum_needs_unit(field, feature, parent):
 
     return None
 
+description = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 24]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
         return header.capitalize()
     return None
 
+description = tr('Retrieve map overview header string from definitions.')
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 25]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
...
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def map_overview_header_element(feature, parent):
     """Retrieve map overview header string from definitions."""
     _ = feature, parent  # NOQA
     header = map_overview_header['string_format']
     return header.capitalize()
 
+description = tr('Retrieve population chart header string from definitions.')
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 26]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
...
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def population_chart_header_element(feature, parent):
     """Retrieve population chart header string from definitions."""
     _ = feature, parent  # NOQA
     header = population_chart_header['string_format']
     return header.capitalize()
 
+description = tr('Retrieve people section header string from definitions.')
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 27]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
...
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def people_section_header_element(feature, parent):
     """Retrieve people section header string from definitions."""
     _ = feature, parent  # NOQA
     header = people_section_header['string_format']
     return header.capitalize()
 
+description = tr('Retrieve age gender section header string from definitions.')
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 28]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
...
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def age_gender_section_header_element(feature, parent):
     """Retrieve age gender section header string from definitions."""
     _ = feature, parent  # NOQA
     header = age_gender_section_header['string_format']
     return header.capitalize()
 
+description = tr('Retrieve age gender section notes string from definitions.')
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 29]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
...
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def age_gender_section_notes_element(feature, parent):
     """Retrieve age gender section notes string from definitions."""
     _ = feature, parent  # NOQA
     notes = age_gender_section_notes['string_format']
     return notes.capitalize()
 
+description = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 30]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
...
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def vulnerability_section_header_element(feature, parent):
     """Retrieve vulnerability section header string from definitions."""
     _ = feature, parent  # NOQA
     header = vulnerability_section_header['string_format']
     return header.capitalize()
 
+description = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 31]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
...
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def female_vulnerability_section_header_element(feature, parent):
     """Retrieve female vulnerability section header string from definitions."""
     _ = feature, parent  # NOQA
     header = female_vulnerability_section_header['string_format']
     return header.capitalize()
 
+description = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 32]
```diff
@@ -129,81 +174,148 @@ def infographic_header_element(impact_function_name, feature, parent):
...
-    args='auto', group=group, usesGeometry=False, referencedColumns=[])
+    args='auto', group=group, usesGeometry=False, referencedColumns=[],
+    help_text=help_message.to_html(), helpText=help_message.to_html())
 def minimum_needs_section_header_element(feature, parent):
     """Retrieve minimum needs section header string from definitions."""
     _ = feature, parent  # NOQA
     header = minimum_needs_section_header['string_format']
     return header.capitalize()
 
+description = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

File: safe/report/expressions/infographic.py [hunk 33]
```diff
@@ -212,9 +324,17 @@ def additional_minimum_needs_section_header_element(feature, parent):
     header = additional_minimum_needs_section_header['string_format']
     return header.capitalize()
 
+description = tr(
```
Comment (ismailsunni) [inline safe/report/expressions/infographic.py]: __[flake8]__
* [E305] expected 2 blank lines after class or function definition, found 1

**Comment by timlinux** [comment]
@gustry see my notes in Gitter about resetting your develop branch. I think you will need to reset and then cherry pick your changes onto the reset branch and then force push your branch to GitHub.

**Comment by Gustry** [comment]
I think I just lost all my work :(

**Comment by Gustry** [comment]
I definitely lost my work. I'm closing this PR and I'm going to do these missing commits again

**Pull request closed without merging**